### PR TITLE
Add first E2E test and changes to the framework required

### PR DESCRIPTION
**Pull request by davidz627**
New E2E test that creates a disk, attaches, mounts, writes to mounted location, unmount, mount somewhere else, reads file, then cleans up.

### CODE REVIEW

**Comment by davidz627** [comment]
@msau42 could you review whenever you have some extra time, thanks!

File: Makefile.orig [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2018 The Kubernetes Authors.
```
Comment (msau42) [inline Makefile.orig]: What is this file for?

File: test/e2e/gce_pd_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2018 Google Inc.
```
Comment (msau42) [inline test/e2e/gce_pd_e2e_test.go]: fix

Comment (davidz627) [reply test/e2e/gce_pd_e2e_test.go]: done

**Comment by davidz627** [comment]
I will merge this for now because I need it to test CI. Feel free to do a post-merge review and I'll compile any comments into a new PR

**Comment by msau42** [comment]
/lgtm
Thanks!

**Pull request merged**